### PR TITLE
Start front-end js section

### LINKED ISSUE #54: Don't bind events on body, especially using unnamed function

### LINKED ISSUE #57: Use data attributes as selectors for interactivity
Don't use classes or IDs. This provides a clean separation of style and interactivity.

### LINKED ISSUE #58: Visual state should be managed by toggling classes
Don't use `.show()` or '.hide()`.

### PR DESCRIPTION
**Pull request by austinstarin**
Close #58, Close #57, Close #54

### CODE REVIEW

File: javascript.md [hunk 1]
```diff
@@ -1,11 +1,56 @@
...
+<a name="data-attributes"></a><a name="1.2"></a>
+[1.2](#data-attributes) Use data attributes as event handlers
+
+> Why? Provides a separation of concerns.
+
+```javascript
+// Good
+$('[data-foo]').removeClass('is-active');
+// Bad
+$('foo').removeClass('is-active');
```
Comment (abea) [inline javascript.md]: Stylistic thing. I'd say we use `$('.foo')` as a more realistic example. A class might also look more distinct from data attributes.

File: javascript.md [hunk 2]
```diff
@@ -1,11 +1,56 @@
...
-Let's start with the basics.
+## Table of Contents
 
-## ESLint
-Use P'unk Ave's [ESLint configuration](https://www.npmjs.com/package/eslint-config-punkave) (based on JavaScript Standard Style) to lint all JavaScript with your editor or during a build process. Install [Atom's ESLint plugin](https://github.com/AtomLinter/linter-eslint).
+1. [Front-end](#front-end)
+
+## Front-end
+<a name="visual-state"></a><a name="1.1"></a>
+[1.1](#visual-state) Add and remove classes to manage visual state
```
Comment (abea) [inline javascript.md]: Can we use H3s instead?

**Comment by abea** [comment]
Content looks good to kick this off. Just a few stylistic notes.

File: javascript.md (hunk 2 above)
Comment (austinstarin) [inline javascript.md]: Let's adjust these across every section in a separate issue. See https://github.com/punkave/best-practices/issues

**Pull request merged**